chapters/grid-search.qmd

```diff
@@ -0,0 +1,390 @@
+
+bst_rnd_res <- 
+  bst_wflow %>% 
+  tune_grid(
+    resamples = sim_rs,
+    metrics = metric_set(brier_class),
+    grid = bst_random %>% select(-grid)
+  )
+bst_rnd_best <- format_best_bst(bst_rnd_res, metric = "brier_class")
+
+bst_sfd_res <- 
+  bst_wflow %>% 
+  tune_grid(
+    resamples = sim_rs,
+    metrics = metric_set(brier_class),
+    grid = bst_sfd %>% select(-grid)
+  )   
+bst_sfd_best <- format_best_bst(bst_sfd_res, metric = "brier_class")
+bst_sfd_brier <- signif(show_best(bst_sfd_res, metric = "brier_class", n = 1)$mean, 3)
+bst_sfd_brier_se <- signif(show_best(bst_sfd_res, metric = "brier_class", n = 1)$std_err, 3)
+
+bst_irr_grid_res <- 
+  collect_metrics(bst_rnd_res) %>% 
+  mutate(grid = "Random") %>% 
+  bind_rows(
+    collect_metrics(bst_sfd_res) %>% 
+      mutate(grid = "Space-Filling")
+  ) %>% 
+  mutate(`learn_rate (log-10)` = log10(learn_rate)) %>% 
+  select(-learn_rate) %>% 
+  pivot_longer(
+    c(trees, `learn_rate (log-10)`),
+    names_to = "parameter",
+    values_to = "value"
+  ) 
+```
+
+We’ll use the simulated training set in @sec-complexity with the same 10-fold cross-validation scheme described there. Once again, Brier scores were used to measure how well each of the `r nrow(bst_reg)`  configurations of boosted tree model predicted the data. @fig-bst-regular-grid-tuning shows the results: a single tree is a poor choice (due to underfitting), and there are several learning rate values that work well. Even though this is not a diverse set of candidate values, we can probably pick out a reasonable candidate with a small Brier score, such as `r bst_reg_best` (although there are a few other candidates that would be good choices). 
+
+```{r}
+#| label: fig-bst-regular-grid-tuning
+#| echo: false
+#| out-width: 50%
+#| fig-width: 4
+#| fig-height: 3.5
+#| fig-cap: "The boosted tree tuning results for the regular grid shown in @fig-bst-grids."
+
+autoplot(bst_reg_res)  +
+  labs(y = "Brier Score") +
+  theme(legend.position = "top")
+```
+
+The pattern shown in this visualization is interesting: the trajectory for the number of trees is different for different values of the learning rate. The large two learning rates are similar, showing that the optimal value is in the mid- to low-range of our grid. The results for the smallest learning rate indicates that better performance might be found using _more_ trees than we used in our grid. This indicates that there is an _interaction effect_ in our tuning parameters (just as we saw for predictors in @sec-interactions). This might not affect how we select the best candidate value for this grid, but it does help build some intuition for this particular model that might come in handy when you tune future models. 
+
+Gaining intituion is much easier for regular grids than other designs since we have a full set of combinations. As we’ll see shortly, this interaction is very hard to see for a space-filling design. 
```

Comment:
```suggestion
Gaining intuition is much easier for regular grids than other designs since we have a full set of combinations. As we’ll see shortly, this interaction is very hard to see for a space-filling design. 
```